src/lib/Microsoft.Health.Extensions.Fhir.R4/Telemetry/Metrics/FhirClientMetrics.cs

```diff
@@ -0,0 +1,38 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.Health.Common.Telemetry;
+
+namespace Microsoft.Health.Extensions.Fhir.Telemetry.Metrics
+{
+    public class FhirClientMetrics
+    {
+        private static string _nameDimension = DimensionNames.Name;
+        private static string _categoryDimension = DimensionNames.Category;
+        private static string _errorTypeDimension = DimensionNames.ErrorType;
+        private static string _errorSeverityDimension = DimensionNames.ErrorSeverity;
+        private static string _operationDimension = DimensionNames.Operation;
+
+        /// <summary>
+        /// A metric recorded when there is an error reading from or connecting with a FHIR server.
+        /// </summary>
+        /// <param name="exceptionName">The name of the exception</param>
+        /// <param name="connectorStage">The stage of the connector</param>
+        public static Metric HandledException(string exceptionName, string connectorStage)
+        {
+            return new Metric(
+                exceptionName,
+                new Dictionary<string, object>
+                {
+                    { _nameDimension, exceptionName },
+                    { _categoryDimension, Category.Errors },
+                    { _errorTypeDimension, ErrorType.GeneralError },
+                    { _errorSeverityDimension, ErrorSeverity.Critical },
```

Comment:
Similarly to the EventHub error metric, should these all be critical too? For example, if the FHIR server returned a 429 because the customer had an unusual amount of data, could eventually correct itself. I'm also wondering if the FHIR server returns 404s for Device or Patients that are not found, which if the connector is set up to Create, this is a normal condition.

Reply:
Yeah, I suppose 429 is more informational than critical. I can update that,

I did some testing (correct me if this is totally wrong) and believe the FHIR server returns a 200 when the Device or Patient are not found with an empty bundle instead of a 404, so I think we don't need to worry about any errors getting logged in that particular case.

Reply:
You might be correct about the 200. It depends on whether or not the FHIR request is a search or a full path to a resource. In other words, if a search is performed with an identifier, it would return an empty bundle. However, if the url is constructed with the FHIR type and FHIR id, I believe the result will be a 404.

I'm not sure that we ever use the FHIR id and only search.

Reply:
It is a 200 now because we do a search not a get since we need to look up by an identifier instead of the internal id.

A 404 could happen if the resource was deleted while we have it cached but that should trigger a retry.